DocsCode/NavigationTutorial/NavigationTutorialViewController.swift

```diff
@@ -70,14 +61,30 @@ class ViewController: UIViewController, MGLMapViewDelegate {
         let routeOptions = NavigationRouteOptions(waypoints: [origin, destination], profileIdentifier: .automobileAvoidingTraffic)
 
         // Generate the route object and draw it on the map
-        _ = Directions.shared.calculate(routeOptions) { [weak self] (session, result) in
-            guard case let .success(response) = result, let route = response.routes?.first, let strongSelf = self else {
-                return
+        Directions.shared.calculate(routeOptions) { [weak self] (session, result) in
+            switch result {
+            case .failure(let error):
+                print(error.localizedDescription)
+            case .success(let response):
+                guard let route = response.routes?.first, let strongSelf = self else {
```

Comment:
nit:
Swift 4.2+ allows you to write
```suggestion
                guard let route = response.routes?.first, let self = self else {
```